File: TreeViewAdvDemo/C#/TreeViewAdvSearchDemo.cs
using System;
using System.Windows.Forms;
using Syncfusion.Windows.Forms;

namespace TreeViewAdvSearchDemo
{
    public partial class TreeViewAdvSearchDemo : MetroForm
    {
        //Initialize the TreeViewSearchForm
        TreeViewSearchForm form;

        #region Constructor

        public TreeViewAdvSearchDemo()
        {
            InitializeComponent();
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.Sizable;
            this.MetroColor = System.Drawing.Color.FromArgb(((int)(((byte)(17)))), ((int)(((byte)(158)))), ((int)(((byte)(218)))));
            this.WindowState = FormWindowState.Normal;
            this.StartPosition = FormStartPosition.CenterScreen;
            this.Padding = new System.Windows.Forms.Padding(10);
            this.MinimumSize = this.Size;
            this.treeViewAdv1.SelectionMode = Syncfusion.Windows.Forms.Tools.TreeSelectionMode.MultiSelectAll;
            //To assign the TreeViewAdv to TreeViewSearchForm
            form = new TreeViewSearchForm(this.treeViewAdv1);
            this.treeViewAdv1.PreviewKeyDown += new PreviewKeyDownEventHandler(treeViewAdv1_PreviewKeyDown);
            this.treeViewAdv1.HideSelection = false;
            this.richTextBox1.AppendText("\n");
            this.treeViewAdv1.OnNodeAfterFound += new Syncfusion.Windows.Forms.Tools.TreeViewOnAfterFindArgs(treeViewAdv1_OnNodeAfterFound);
            this.treeViewAdv1.OnNodeBeforeFind += new Syncfusion.Windows.Forms.Tools.TreeViewOnBeforeFindArgs(treeViewAdv1_OnNodeBeforeFind);
            this.treeViewAdv1.OnNodeReplacing += new Syncfusion.Windows.Forms.Tools.TreeViewOnReplacingArgs(treeViewAdv1_OnNodeReplacing);
            this.treeViewAdv1.OnNodeReplaced += new Syncfusion.Windows.Forms.Tools.TreeViewOnReplacedArgs(treeViewAdv1_OnNodeReplaced);
        }

        #endregion

        void treeViewAdv1_OnNodeAfterFound(object sender, Syncfusion.Windows.Forms.Tools.TreeNodeAdvAfterFindArgs e)
        {
            this.richTextBox1.AppendText("\n AfterNodeMatchFound Event Triggered");
        }

        void treeViewAdv1_OnNodeReplaced(object sender, Syncfusion.Windows.Forms.Tools.TreeNodeAdvOnReplacedArgs e)
        {
            this.richTextBox1.AppendText("\n OnNodeTextReplaced Event Triggered");
        }

        void treeViewAdv1_OnNodeReplacing(object sender, Syncfusion.Windows.Forms.Tools.TreeNodeAdvOnReplacingArgs e)
        {
            this.richTextBox1.AppendText("\n OnNodeTextReplacing Event Triggered");
        }

        void treeViewAdv1_OnNodeBeforeFind(object sender, Syncfusion.Windows.Forms.Tools.TreeNodeAdvBeforeFindArgs e)
        {
            this.richTextBox1.AppendText("\n BeforeNodeMatchFound Event Triggered");
        }

        //To show the search dialog
        void treeViewAdv1_PreviewKeyDown(object sender, PreviewKeyDownEventArgs e)
        {
            if (e.Modifiers == Keys.Control && e.KeyValue == (Char)Keys.F)
            {
                form = new TreeViewSearchForm(this.treeViewAdv1);
                form.ShowDialog();
            }
        }
    }
}

File: TreeViewAdvDemo/VB/TreeViewAdvSearchDemo.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms

Namespace TreeViewAdvSearchDemo
    Partial Public Class TreeViewAdvSearchDemo
        Inherits MetroForm
        'Initialize the TreeViewSearchForm
        Private form As TreeViewSearchForm

#Region "Constructor"

        Public Sub New()
            InitializeComponent()
            Me.FormBorderStyle = FormBorderStyle.Sizable
            Me.MetroColor = System.Drawing.Color.FromArgb((CInt(Fix((CByte(17))))), (CInt(Fix((CByte(158))))), (CInt(Fix((CByte(218))))))
            Me.WindowState = FormWindowState.Normal
            Me.StartPosition = FormStartPosition.CenterScreen
            Me.Padding = New Padding(10)
            Me.MinimumSize = Me.Size
            Me.treeViewAdv1.SelectionMode = Tools.TreeSelectionMode.MultiSelectAll
            'To assign the TreeViewAdv to TreeViewSearchForm
            form = New TreeViewSearchForm(Me.treeViewAdv1)
            AddHandler treeViewAdv1.PreviewKeyDown, AddressOf treeViewAdv1_PreviewKeyDown
            Me.treeViewAdv1.HideSelection = False
            Me.richTextBox1.AppendText(Constants.vbLf)
            AddHandler treeViewAdv1.OnNodeAfterFound, AddressOf treeViewAdv1_OnNodeAfterFound
            AddHandler treeViewAdv1.OnNodeBeforeFind, AddressOf treeViewAdv1_OnNodeBeforeFind
            AddHandler treeViewAdv1.OnNodeReplacing, AddressOf treeViewAdv1_OnNodeReplacing
            AddHandler treeViewAdv1.OnNodeReplaced, AddressOf treeViewAdv1_OnNodeReplaced
        End Sub

#End Region

        Private Sub treeViewAdv1_OnNodeAfterFound(ByVal sender As Object, ByVal e As Tools.TreeNodeAdvAfterFindArgs)
            Me.richTextBox1.AppendText(Constants.vbLf & " AfterNodeMatchFound Event Triggered")
        End Sub

        Private Sub treeViewAdv1_OnNodeReplaced(ByVal sender As Object, ByVal e As Tools.TreeNodeAdvOnReplacedArgs)
            Me.richTextBox1.AppendText(Constants.vbLf & " OnNodeTextReplaced Event Triggered")
        End Sub

        Private Sub treeViewAdv1_OnNodeReplacing(ByVal sender As Object, ByVal e As Tools.TreeNodeAdvOnReplacingArgs)
            Me.richTextBox1.AppendText(Constants.vbLf & " OnNodeTextReplacing Event Triggered")
        End Sub

        Private Sub treeViewAdv1_OnNodeBeforeFind(ByVal sender As Object, ByVal e As Tools.TreeNodeAdvBeforeFindArgs)
            Me.richTextBox1.AppendText(Constants.vbLf & " BeforeNodeMatchFound Event Triggered")
        End Sub

        'To show the search dialog
        Private Sub treeViewAdv1_PreviewKeyDown(ByVal sender As Object, ByVal e As PreviewKeyDownEventArgs)
            If e.Modifiers = Keys.Control AndAlso e.KeyValue = Keys.F Then
                form = New TreeViewSearchForm(Me.treeViewAdv1)
                form.ShowDialog()
            End If
        End Sub
    End Class
End Namespace